server/src/main/java/org/eclipse/lsp/cobol/domain/databus/impl/DefaultDataBusBroker.java

```diff
@@ -15,64 +15,46 @@
 
 package org.eclipse.lsp.cobol.domain.databus.impl;
 
-import org.eclipse.lsp.cobol.domain.databus.model.RegistryId;
-import org.eclipse.lsp.cobol.domain.event.api.EventObserver;
-import org.eclipse.lsp.cobol.domain.event.model.DataEvent;
-import org.eclipse.lsp.cobol.domain.event.model.DataEventType;
-import org.eclipse.lsp.cobol.service.utils.CustomThreadPoolExecutor;
+import com.google.common.eventbus.AsyncEventBus;
+import com.google.common.eventbus.EventBus;
 import com.google.inject.Inject;
 import com.google.inject.Singleton;
 import lombok.NonNull;
 import lombok.extern.slf4j.Slf4j;
+import org.eclipse.lsp.cobol.domain.databus.api.DataBusBroker;
+import org.eclipse.lsp.cobol.domain.databus.api.DeadEventSubscriber;
+import org.eclipse.lsp.cobol.domain.databus.model.DataEvent;
+import org.eclipse.lsp.cobol.service.utils.CustomThreadPoolExecutor;
+
+import java.util.concurrent.ExecutorService;
 
 /**
  * This class is the default implementation for databus broker.
- *
- * @param <T> - a data event class managed by the implementation
- * @param <S> - a subscriber class for this event
  */
 @Slf4j
 @Singleton
-public class DefaultDataBusBroker<T extends DataEvent, S> extends AbstractDataBusBroker<T, S> {
+public class DefaultDataBusBroker implements DataBusBroker {
```

Comment:
I would recommend adding debug logging to the methods

Reply:
Done.